### PR TITLE
Todo-17-FE: As a user, I want to select all todos instantly and get warning when trying to delete anything

### PR DESCRIPTION
**Pull request by itc02**
image
image

### CODE REVIEW

File: src/components/dialogs/deleteItems/styles.tsx [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import styled from 'styled-components';
+
+export const Question = styled.div`
+  font-size: 20px;
```
Comment (arpinehp) [inline src/components/dialogs/deleteItems/styles.tsx:4]: use `rem`

Comment (itc02) [reply src/components/dialogs/deleteItems/styles.tsx:4]: I created separate ticket for it

File: src/components/app/App.tsx [hunk 2]
```diff
@@ -1,10 +1,12 @@
 import React from 'react';
 import { MainTable } from '../mainTable/MainTable';
-import './App.css';
+import { Provider } from '../../utils/globalState/Provider';
 
 const App: React.FC = () => {
   return (
-   <MainTable />
+    <Provider>
+        <MainTable />
```
Comment (arpinehp) [inline src/components/app/App.tsx]: 2 spaces instead of 4

File: src/components/mainTable/MainTable.tsx [hunk 3]
```diff
@@ -102,6 +113,7 @@ export const MainTable:React.FC = () => {
   }, [ currentPage, currentPer, order, searchString ]);
 
   return (
+    <div style={{backgroundColor: '#ececec', height: '100%'}}>
```
Comment (arpinehp) [inline src/components/mainTable/MainTable.tsx:116]: ??

File: src/utils/globalState/Provider.tsx [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import React, { useReducer } from "react";
+import { initialState } from './initialState';
+import reducer from './reducer';
+import { StateType } from './initialState';
+import { Actions } from './actions';
+import { Context } from './useGlobalState';
+
+
+
```
Comment (arpinehp) [inline src/utils/globalState/Provider.tsx:9]: remove spaces

File: src/utils/globalState/reducer.ts [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+      return {
+        selectedTodos,
+        selectedUsers: action.arrayOfIds
+      }
+    default:
+      throw new Error('Unknown action: ' + action);
+  }
+}
+
+export default reducer;
```
Comment (arpinehp) [inline src/utils/globalState/reducer.ts:53]: :+1:

**Review by arpinehp — APPROVED** [review]

**Pull request merged**